### PR TITLE
Add Polish damaging and goodfaith

### PR DESCRIPTION
**Pull request by Ladsgroup**
Damaging:

```
ScikitLearnClassifier
 - type: GradientBoosting
 - params: max_leaf_nodes=null, min_samples_split=2, scale=true, balanced_sample=false, subsample=1.0, presort="auto", min_samples_leaf=1, verbose=0, min_weight_fraction_leaf=0.0, learning_rate=0.01, max_depth=5, balanced_sample_weight=true, init=null, max_features="log2", center=true, n_estimators=700, random_state=null, warm_start=false, loss="deviance"
 - version: 0.1.1
 - trained: 2016-07-03T21:30:13.453169

Table:
             ~False    ~True
    -----  --------  -------
    False      4064      222
    True         20       52

Accuracy: 0.944
Precision: 0.19
Recall: 0.722
PR-AUC: 0.291
ROC-AUC: 0.931
Recall @ 0.1 false-positive rate: threshold=0.972, recall=0.014, fpr=0.0
Filter rate @ 0.9 recall: threshold=0.11, filter_rate=0.783, recall=0.903
Filter rate @ 0.75 recall: threshold=0.382, filter_rate=0.924, recall=0.75
```

Goodfaith:

```
ScikitLearnClassifier
 - type: GradientBoosting
 - params: min_weight_fraction_leaf=0.0, max_depth=3, warm_start=false, n_estimators=700, init=null, random_state=null, max_features="log2", max_leaf_nodes=null, min_samples_split=2, min_samples_leaf=1, learning_rate=0.01, balanced_sample=false, balanced_sample_weight=true, scale=true, subsample=1.0, presort="auto", loss="deviance", center=true, verbose=0
 - version: 0.1.1
 - trained: 2016-07-04T02:46:47.484697

Table:
             ~False    ~True
    -----  --------  -------
    False        26        3
    True        213     4116

Accuracy: 0.95
Precision: 0.999
Recall: 0.951
PR-AUC: 1.0
ROC-AUC: 0.979
Recall @ 0.1 false-positive rate: threshold=0.034, recall=1.0, fpr=0.007
Filter rate @ 0.9 recall: threshold=0.741, filter_rate=0.105, recall=0.9
Filter rate @ 0.75 recall: threshold=0.886, filter_rate=0.255, recall=0.75
```

### CODE REVIEW

**Comment by halfak** [comment]
Looks like we didn't have enough positive observations to do a good test. 

Also, it looks like the merge of labeled edits with unabled edits isn't quite right.  This is the first campaign with a 2500/2500 split of needs-review/not.  

I was imagining that we'd over-sample this a bit to make it representative.  First, we worked with 500k revisions to get the necessary 2500 that need review.  See https://meta.wikimedia.org/wiki/Research_talk:Revision_scoring_as_a_service/Work_log/2016-02-23

Based on that work-log, it looks like 16.5% of revisions need review.  I'd like to sample with replacement from the revisions-for-review that did _not_ need review to balance out the sample. `(2500/0.165) - 2500 == 12651`.  This is done by sampling with replacement using `shuf -r -n 12651` below.  

Here's the rule I have in mind. 

```
datasets/plwiki.rev_damaging.15k_2016.tsv: \
        datasets/plwiki.prelabeled_revisions.15k_2015.tsv
    (cat datasets/plwiki.rev_damaging.5k_2016.tsv | grep True; \
     cat datasets/plwiki.rev_damaging.5k_2016.tsv | grep False | shuf -r -n 12651;) | \
    shuf > \
    datasets/plwiki.rev_damaging.15k_2016.tsv
```

**Comment by Ladsgroup** [comment]
Oh, I check several times and this campaign doesn't look like a half needing review/half doesn't.

File: Makefile [hunk 1]
```diff
@@ -1556,26 +1561,19 @@ datasets/plwiki.rev_damaging.5k_2016.tsv:
...
-		grep False | \
-		cat datasets/plwiki.rev_damaging.5k_2016.tsv - | shuf > \
-	datasets/plwiki.rev_damaging.20k_2016.tsv
-
-datasets/plwiki.features_damaging.20k_2016.tsv: \
-		datasets/plwiki.rev_damaging.20k_2016.tsv
-	cat datasets/plwiki.rev_damaging.20k_2016.tsv | \
+datasets/plwiki.features_damaging.5k_2016.tsv: \
+		datasets/plwiki.rev_damaging.5k_2016.tsv
+	cat datasets/plwiki.rev_damaging.5k_2016.tsv | \
```
Comment (halfak) [inline Makefile]: Should be rev_damaging.15k_2016.

File: Makefile [hunk 2]
```diff
@@ -1556,26 +1561,19 @@ datasets/plwiki.rev_damaging.5k_2016.tsv:
...
 	revscoring extract_features \
 		editquality.feature_lists.plwiki.damaging \
 		--host https://pl.wikipedia.org \
 		--include-revid \
 		--verbose > \
-	datasets/plwiki.features_damaging.20k_2016.tsv
+	datasets/plwiki.features_damaging.5k_2016.tsv
```
Comment (halfak) [inline Makefile]: Still features_damaging.15k_2016

**Comment by Ladsgroup** [comment]
```
(p3)ladsgroup@ores-compute-01:~/editquality$ make models/plwiki.damaging.rf.model
cut datasets/plwiki.features_damaging.15k_2016.tsv -f2- | \
revscoring train_test \
    revscoring.scorer_models.RF \
    editquality.feature_lists.plwiki.damaging \
    --version 0.1.0 \
    -p 'max_features="log2"' \
    -p 'criterion="entropy"' \
    -p 'min_samples_leaf=1' \
    -p 'n_estimators=640' \
    -s 'table' -s 'accuracy' -s 'precision' -s 'recall' -s 'pr' -s 'roc' -s 'recall_at_fpr(max_fpr=0.10)' -s 'filter_rate_at_recall(min_recall=0.90)' -s 'filter_rate_at_recall(min_recall=0.75)' \
    --balance-sample-weight \
    --center --scale \
    --label-type=bool > \
models/plwiki.damaging.rf.model
2016-07-09 05:58:01,986 INFO:revscoring.utilities.train_test -- Training model...
2016-07-09 05:58:20,198 INFO:revscoring.utilities.train_test -- Testing model...
ScikitLearnClassifier
 - type: RF
 - params: center=true, verbose=0, balanced_sample_weight=true, max_features="log2", warm_start=false, max_leaf_nodes=null, n_estimators=640, min_weight_fraction_leaf=0.0, class_weight=null, min_samples_split=2, balanced_sample=false, min_samples_leaf=1, oob_score=false, n_jobs=1, scale=true, criterion="entropy", bootstrap=true, max_depth=null, random_state=null
 - version: 0.1.0
 - trained: 2016-07-09T05:58:20.195165

Table:
             ~False    ~True
    -----  --------  -------
    False      2930        1
    True         57       25

Accuracy: 0.981
Precision: 0.962
Recall: 0.305
PR-AUC: 0.676
ROC-AUC: 0.979
Recall @ 0.1 false-positive rate: threshold=0.4, recall=0.378, fpr=0.088
Filter rate @ 0.9 recall: threshold=0.048, filter_rate=0.906, recall=0.902
Filter rate @ 0.75 recall: threshold=0.108, filter_rate=0.951, recall=0.756
```

**Comment by Ladsgroup** [comment]
good faith:

```
2016-07-09 07:12:59,678 INFO:revscoring.utilities.train_test -- Training model...
2016-07-09 07:13:06,673 INFO:revscoring.utilities.train_test -- Testing model...
ScikitLearnClassifier
 - type: RF
 - params: min_samples_leaf=1, balanced_sample_weight=true, class_weight=null, min_weight_fraction_leaf=0.0, bootstrap=true, warm_start=false, criterion="entropy", verbose=0, scale=true, max_features="log2", max_leaf_nodes=null, n_jobs=1, max_depth=null, balanced_sample=false, random_state=null, n_estimators=320, min_samples_split=2, center=true, oob_score=false
 - version: 0.1.0
 - trained: 2016-07-09T07:13:06.670661

Table:
             ~False    ~True
    -----  --------  -------
    False        14       22
    True          0     2975

Accuracy: 0.993
Precision: 0.993
Recall: 1.0
PR-AUC: 1.0
ROC-AUC: 0.966
Recall @ 0.1 false-positive rate: threshold=0.5, recall=1.0, fpr=0.008
Filter rate @ 0.9 recall: threshold=0.991, filter_rate=0.105, recall=0.905
Filter rate @ 0.75 recall: threshold=1.0, filter_rate=0.193, recall=0.816
```

**Comment by halfak** [comment]
Great teamwork :D

**Pull request merged**